advanced_alchemy/extensions/sanic.py

```diff
@@ -96,6 +97,14 @@ def __init__(
             else self.sqlalchemy_config.get_engine()
         )
         self.session_maker = self.sqlalchemy_config.create_session_maker()
+
+        if self.sqlalchemy_config.session_config.class_ is not Empty:
+            self.session_class = self.sqlalchemy_config.session_config.class_
+        elif isinstance(self.session_maker, async_sessionmaker):
+            self.session_class = AsyncSession
+        else:
+            self.session_class = Session
```

Comment:
I don't think this is exactly right.. a session maker can create an async session.  However, I don't know a better way to determine this right now.

Reply:
Ah, wait!  The session maker has the `class_` defined, I think..  In theory, that should ALWAYS have the correct value.  Bonus: We don't have to move imports around.

```suggestion
        else:
            self.session_class = self.session_maker.class_
```

Reply:
Much better.